x-pack/plugins/cloud_security_posture/server/routes/configuration/update_rules_configuration.ts

```diff
@@ -77,57 +69,197 @@ const getEnabledRulesByBenchmark = (rules: SavedObjectsFindResponse<CspRule>['sa
+  cspRules: Array<SavedObject<PackagePolicyRuleUpdatePayload>>
 ): CspRulesConfiguration => ({
   runtime_cfg: {
-    activated_rules: getEnabledRulesByBenchmark(cspRules.saved_objects),
+    activated_rules: getEnabledRulesByBenchmark(cspRules),
   },
 });
 
-export const convertRulesConfigToYaml = (config: CspRulesConfiguration): string => {
-  return yaml.safeDump(config);
-};
-
-export const setVarToPackagePolicy = (
+/* @internal */
+export const getUpdatedPackagePolicy = (
   packagePolicy: PackagePolicy,
   runtimeCfg: string
-): PackagePolicy => {
-  const configFile: PackagePolicyConfigRecord = {
-    runtimeCfg: { type: 'yaml', value: runtimeCfg },
-  };
-  const updatedPackagePolicy = produce(packagePolicy, (draft) => {
+): PackagePolicy =>
+  produce(packagePolicy, (draft) => {
     unset(draft, 'id');
-    if (draft.vars) {
-      draft.vars.runtimeCfg = configFile.runtimeCfg;
-    } else {
-      draft.vars = configFile;
-    }
+
+    if (!draft.vars) draft.vars = {};
+    draft.vars.runtimeCfg = { type: 'yaml', value: runtimeCfg };
   });
-  return updatedPackagePolicy;
-};
 
-export const updateAgentConfiguration = async (
-  packagePolicyService: PackagePolicyServiceInterface,
-  packagePolicy: PackagePolicy,
-  esClient: ElasticsearchClient,
+/**
+ * gets all rules of a package policy with fields required for runtime config
+ */
+const getAllPackagePolicyCspRules = (
   soClient: SavedObjectsClientContract,
-  user: AuthenticatedUser | null
-): Promise<PackagePolicy> => {
-  const cspRules = await getCspRules(soClient, packagePolicy);
-  const rulesConfig = createRulesConfig(cspRules);
-  const runtimeCfg = convertRulesConfigToYaml(rulesConfig);
-  const updatedPackagePolicy = setVarToPackagePolicy(packagePolicy, runtimeCfg);
-  const options = { user: user ? user : undefined };
+  packagePolicyId: PackagePolicy['id'],
+  policyId: PackagePolicy['policy_id']
+) =>
+  soClient.find<PackagePolicyRuleUpdatePayload>({
+    type: CSP_RULE_SAVED_OBJECT_TYPE,
+    filter: createCspRuleSearchFilterByPackagePolicy({
+      packagePolicyId,
+      policyId,
+    }),
+    fields: RUNTIME_CFG_FIELDS,
+    searchFields: ['name'],
+    perPage: 10000,
+  });
+
+/**
+ * gets rules by id of a package policy with fields required for updating SO object
+ */
+const getPartialCspRulesById = (
+  soClient: CspApiRequestHandlerContext['soClient'],
+  ruleIds: string[]
+) =>
+  soClient.bulkGet<SavedObjectRuleUpdatePayload>(
+    ruleIds.map((ruleId) => ({
+      id: ruleId,
+      type: CSP_RULE_SAVED_OBJECT_TYPE,
+      fields: ['enabled'],
+    }))
+  );
+
+const getRuntimeCfgVarValue = (rules: Array<SavedObject<PackagePolicyRuleUpdatePayload>>) =>
+  yaml.safeDump(createRulesConfig(rules));
+
+/**
+ * Updates the package policy vars object with a new value for the runtimeCfg key
+ * @internal
+ * */
+export const updatePackagePolicyVars = async ({
+  packagePolicyService,
+  packagePolicy,
+  esClient,
+  soClient,
+  user,
+}: {
```

Comment:
nit: i would consider pass the `packagePolicyRulesSO` as argument instead of fetching it inside the update function

Reply:
`updatePackagePolicyVars` is called twice, once from the API and once on package post-create.

if we pass it as argument we'd need to fetch in both of these before passing it, i wanted to avoid it

Reply:
my bad, we only need to fetch it once when updating the config. on post create we already have it. 

lifted the policy fetching outside of this function to the route handler. if it doesn't exists, we don't go into the update sequence at all. good point!